### DISCUSSION TITLE
Contract metadata from the compiler

### DISCUSSION THREAD

**Discussion by numpde** [NomicFoundation/hardhat#4022] [Troubleshooting]
I saw in the bytecode deployed via deployProxy that an IPFS hash of the metadata is appended, presumably according to [this spec](https://docs.soliditylang.org/en/latest/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode). Where can I find the exact "canonical metadata" that was used for it (so I could upload it to IPFS)?

And how could I avoid appending it?

**Comment by fvictorio** [comment]
I can't answer the first question because that's handled by the solidity compiler and I ignore the details. But in the section of the docs you linked there is a bit more information, including a link to the Metadata Playground that maybe you can use to understand better how it's computed. Alternatively, you can ask in the [solidity forum](https://forum.soliditylang.org/).

> And how could it avoid appending it?

If you are using 0.8.18 or later, you can set the `metadata.appendCBOR` option to false:

```js
module.exports = {
  solidity: {
    version: "0.8.18",
    settings: {
      metadata: {
        appendCBOR: false
      }
    }
  },
};
```

**Reply by numpde** [reply]
Thank you for the reply. 

Meanwhile, I found that **artifacts/build-info/*.json** contains the said metadata for each contract and I was able to reproduce the IPFS hash from it (as seen on the metadata playground). I made [a tool for that](https://numpde.github.io/ipfs-for-text/), fwiw.
